packages/stacking/src/index.ts

```diff
@@ -341,6 +519,144 @@ export class StackingClient {
     return txOptions;
   }
 
+  getDelegateOptions({
+    contract,
+    amountMicroStx,
+    delegateTo,
+    untilBurnBlockHeight,
+    poxAddress,
+  }: {
+    contract: string;
+    amountMicroStx: BN;
+    delegateTo: string;
+    untilBurnBlockHeight?: number;
+    poxAddress?: string;
+  }) {
+    let address = undefined;
+
+    if (poxAddress) {
+      const { hashMode, data } = decodeBtcAddress(poxAddress);
+      const hashModeBuffer = bufferCV(new BN(hashMode, 10).toBuffer());
+      const hashbytes = bufferCV(data);
+      address = someCV(
+        tupleCV({
+          hashbytes,
+          version: hashModeBuffer,
+        })
+      );
+    }
+
+    const [contractAddress, contractName] = contract.split('.');
+    const network = this.network;
+    const txOptions: ContractCallOptions = {
+      contractAddress,
+      contractName,
+      functionName: 'delegate-stx',
+      functionArgs: [
+        uintCV(amountMicroStx.toString(10)),
+        standardPrincipalCV(delegateTo),
+        untilBurnBlockHeight ? uintCV(untilBurnBlockHeight) : noneCV(),
+        address ? address : noneCV(),
+      ],
+      validateWithAbi: true,
+      network,
+    };
+    return txOptions;
+  }
+
+  getDelegateStackOptions({
+    contract,
+    stacker,
+    amountMicroStx,
+    poxAddress,
+    burnBlockHeight,
+    cycles,
+    nonce,
+  }: {
+    contract: string;
+    stacker: string;
+    amountMicroStx: BN;
+    poxAddress: string;
+    burnBlockHeight: number;
+    cycles: number;
+    nonce?: BN;
+  }) {
+    const { hashMode, data } = decodeBtcAddress(poxAddress);
+    const hashModeBuffer = bufferCV(new BN(hashMode, 10).toBuffer());
+    const hashbytes = bufferCV(data);
+    const address = tupleCV({
+      hashbytes,
+      version: hashModeBuffer,
+    });
+
+    const [contractAddress, contractName] = contract.split('.');
+    const network = this.network;
+    const txOptions: ContractCallOptions = {
+      contractAddress,
+      contractName,
+      functionName: 'delegate-stack-stx',
+      functionArgs: [
+        standardPrincipalCV(stacker),
+        uintCV(amountMicroStx.toString(10)),
+        address,
+        uintCV(burnBlockHeight),
+        uintCV(cycles),
+      ],
+      validateWithAbi: true,
+      network,
+    };
+
+    if (nonce) {
+      txOptions.nonce = nonce;
+    }
+
+    return txOptions;
+  }
+
+  getStackAggregationCommitOptions({
+    contract,
+    poxAddress,
+    rewardCycle,
+  }: {
+    contract: string;
+    poxAddress: string;
+    rewardCycle: number;
+  }) {
+    const { hashMode, data } = decodeBtcAddress(poxAddress);
+    const hashModeBuffer = bufferCV(new BN(hashMode, 10).toBuffer());
+    const hashbytes = bufferCV(data);
+    const address = tupleCV({
+      hashbytes,
+      version: hashModeBuffer,
+    });
+
+    const [contractAddress, contractName] = contract.split('.');
```

Comment:
Maybe a new fn can be made to abstract out contract name parsing, something like:


```typescript
function parseContrastName (contract: string) {
  if (stringNotValid)  {
    throw new Error('Unable to parse string error');
  }
  return [address, name] = contract.split('.'); 
}
``

Reply:
You can `import { validateStacksAddress } from '@stacks/transactions'` and use it here. 

We should just have a `validateContractIdentifier` function if we don't already. But don't consider that blocking for this PR.

Reply:
using these suggestions for a new method `parseContractId`. please review again 🙏🏼